src/librustdoc/clean/mod.rs

```diff
@@ -995,7 +991,9 @@ impl<'tcx> Clean<FnDecl> for (DefId, ty::PolyFnSig<'tcx>) {
                     .iter()
                     .map(|t| Argument {
                         type_: t.clean(cx),
-                        name: names.next().map_or_else(|| String::new(), |name| name.to_string()),
+                        name: Symbol::intern(
+                            &names.next().map_or_else(|| String::new(), |name| name.to_string()),
+                        ),
```

Comment:
`fn_arg_names` already returns a Symbol, so no need to stringify it only to intern again.
```suggestion
                        name: names.next().unwrap_or(kw::Invalid),
```